src/app/analyze/stack/stack-details/stack-details.component.ts

```diff
@@ -188,10 +188,10 @@ export class StackDetailsComponent implements OnInit {
     const url: string = this.similarStacks[0].uri;
     this.recoArray[this.currentIndex]['rows'] = [];
     this.recoArray[this.currentIndex]['url'] = url;
-    for (var component in components) {
+    for (let component of components) {
```

Comment:
for .. in returns every object whereas for .. of returns the next iteration in collection type of object.